src/components/Post/Post.js

```diff
@@ -3,12 +3,16 @@ import Checkbox from '@material-ui/core/Checkbox';
 import classes from './Post.module.css';
 
 class Post extends Component {
-    state = {
-        checked: false
+    constructor(props) {
+        super(props);
+        this.state = {
+            checked: false,
+        };
     }
 
     handleChange = () => {
         this.setState(prevState => ({checked: !prevState.checked}));
+        this.props.onChange && this.props.onChange(this.state);
```

Comment:
This will not pass the latest updated state. We can pass a second parameter to setState and call onChange

Reply:
Should it be like this:
 handleChange = () => {
        this.setState(prevState => ({checked: !prevState.checked}), this.props.onChange && this.props.onChange(this.state));
    }